bson/decoder_test.go

```diff
@@ -199,12 +199,12 @@ func TestDecoderv2(t *testing.T) {
 		dc2 := bsoncodec.DecodeContext{Registry: NewRegistryBuilder().Build()}
 		dec, err := NewDecoderWithContext(dc1, bsonrw.NewBSONDocumentReader([]byte{}))
 		noerr(t, err)
-		if dec.dc != dc1 {
+		if !reflect.DeepEqual(dec.dc, dc1) {
```

Comment:
Why did these have to switch to deep equals?

Reply:
You can only directly compare structs in go if they use non-nesting types.  I'm not sure if it's necessary right now but it's much safer than a strong inequality.